components/brave_ads/core/browser/search_result_ad/search_result_ad_converting_util.cc

```diff
@@ -193,155 +193,134 @@ bool SetSearchAdProperty(const schema_org::mojom::PropertyPtr& ad_property,
   return false;
 }
 
-// TODO(https://github.com/brave/brave-browser/issues/25971): Reduce cognitive
-// complexity.
-absl::optional<SearchResultAdMap> ParseSearchResultAdMapEntityProperties(
-    const schema_org::mojom::EntityPtr& entity) {
-  DCHECK(entity);
-  DCHECK_EQ(entity->type, kProductType);
+void ConvertSearchResultAdEntity(const schema_org::mojom::EntityPtr& ad_entity,
```

Comment:
`entity` as this is not necessarily an ad entity, but I could be wrong :-)

Reply:
Yes, it can be not an ad entity. Fixed the discrepancy by renaming the argument to `entity` and the function name to `ConvertEntityToSearchResultAd`.